gapis/api/vulkan/replay.go

```diff
@@ -842,58 +842,37 @@ type profileRequest struct {
 func (a API) GetInitialPayload(ctx context.Context,
 	capture *path.Capture,
 	device *device.Instance,
-	out transform.Writer) error {
-	transforms := transform.Transforms{}
-	transforms.Add(&makeAttachementReadable{false})
-	transforms.Add(&dropInvalidDestroy{tag: "GetInitialPayload"})
+	out transform2.Writer) error {
+
 	initialCmds, im, _ := initialcmds.InitialCommands(ctx, capture)
 	out.State().Allocator.ReserveRanges(im)
+	cmdGenerator := commandGenerator.NewLinearCommandGenerator(initialCmds, nil)
 
-	return transforms.TransformAll(ctx, initialCmds, uint64(len(initialCmds)), out)
-}
-
-func (a API) CleanupResources(ctx context.Context,
-	device *device.Instance,
-	out transform.Writer) error {
-	transforms := transform.Transforms{}
-	transforms.Add(&destroyResourcesAtEOS{})
-	return transforms.TransformAll(ctx, []api.Cmd{}, 0, out)
-}
+	transforms := getCommonInitializationTransforms("GetInitialPayload")
 
-func getCommonInitializationTransforms(tag string) []transform2.Transform {
-	return []transform2.Transform{
-		newMakeAttachmentReadable2(false),
-		newDropInvalidDestroy2(tag),
+	chain := transform2.CreateTransformChain(cmdGenerator, transforms, out)
+	controlFlow := controlFlowGenerator.NewLinearControlFlowGenerator(chain)
+	if err := controlFlow.TransformAll(ctx); err != nil {
+		log.E(ctx, "[GetInitialPayload] Error: %v", err)
+		return err
 	}
-}
 
-func appendLogTransforms(ctx context.Context, tag string, capture *capture.GraphicsCapture, transforms []transform2.Transform) []transform2.Transform {
-	if config.LogTransformsToFile {
-		newTransforms := make([]transform2.Transform, 0)
-		newTransforms = append(newTransforms, newFileLog(ctx, "0_original_cmds"))
-		for i, t := range transforms {
-			var name string
-			if n, ok := t.(interface {
-				Name() string
-			}); ok {
-				name = n.Name()
-			} else {
-				name = strings.Replace(fmt.Sprintf("%T", t), "*", "", -1)
-			}
-			newTransforms = append(newTransforms, t, newFileLog(ctx, fmt.Sprintf("%v_cmds_after_%v", i+1, name)))
-		}
-		transforms = newTransforms
-	}
+	return nil
+}
 
-	if config.LogTransformsToCapture {
-		transforms = append(transforms, newCaptureLog(ctx, capture, tag+"_replay_log.gfxtrace"))
+func (a API) CleanupResources(ctx context.Context, device *device.Instance, out transform2.Writer) error {
```

Comment:
After clarifying where CleanupResources get a State, I would say:
```
// CleanupResources emits commands to destroy the resources of the state
// that is carried in its transform writer argument.
```

Reply:
I am not sure to write explicitly writer argument. 

I would word as 

// CleanupResources emits commands to destroy the resources of the given state

Otherwise we would need to explain everywhere that uses this and I don't think we should explain the arguments because they can be changed in the future and comments may become outdated and confusing without we realize.

The fact that writer carries a state is already documented in the Writer interface.

Reply:
Yeah I hesitated to mention the argument... I think your version is better, go for it :)